### PR TITLE
feat/209: create delete art

### PR DESCRIPTION
**Pull request by NMarteniuk**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Media/Art/DeleteArtHandlerTest.cs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    private void SetupMockLogger(int id, string errorMessage)
+    {
+        _mockLogger.Setup(x => x.LogError(id, errorMessage));
+    }
+
+    private string ArtNotFoundErrorMessage(int id)
+    {
+        return string.Format(
+            ErrorMessages.EntityByIdNotFound,
+            nameof(ArtEntity),
```
Comment (OlhaPivniuk) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Media/Art/DeleteArtHandlerTest.cs:161]: Use typeof(ArtEntity).Name instead  nameof()

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Media/Art/DeleteArtHandlerTest.cs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            id);
+    }
+
+    private string DeleteFailedErrorMessage(int id)
+    {
+        return string.Format(
+            ErrorMessages.DeleteFailed,
+            nameof(ArtEntity),
```
Comment (OlhaPivniuk) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Media/Art/DeleteArtHandlerTest.cs:169]: Use typeof(ArtEntity).Name instead  nameof()

File: Streetcode/Streetcode.BLL/MediatR/Media/Art/Delete/DeleteArtHandler.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        var response = new DeleteArtResponseDto(true);
+
+        return Result.Ok(response);
+    }
+
+    private Result<DeleteArtResponseDto> ArtNotFoundError(int id)
+    {
+        string errorMessage = string.Format(
+            ErrorMessages.EntityByIdNotFound,
+            nameof(ArtEntity),
```
Comment (OlhaPivniuk) [inline Streetcode/Streetcode.BLL/MediatR/Media/Art/Delete/DeleteArtHandler.cs:51]: Use typeof(ArtEntity).Name instead  nameof()

File: Streetcode/Streetcode.BLL/MediatR/Media/Art/Delete/DeleteArtHandler.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        _logger.LogError(id, errorMessage);
+
+        return Result.Fail(errorMessage);
+    }
+
+    private Result<DeleteArtResponseDto> DeleteFailed(int id)
+    {
+        string errorMessage = string.Format(
+            ErrorMessages.DeleteFailed,
+            nameof(ArtEntity),
```
Comment (OlhaPivniuk) [inline Streetcode/Streetcode.BLL/MediatR/Media/Art/Delete/DeleteArtHandler.cs:63]: Use typeof(ArtEntity).Name instead  nameof()

File: Streetcode/Streetcode.WebApi/Controllers/Media/Images/ArtController.cs [hunk 5]
```diff
@@ -32,4 +34,10 @@ public async Task<IActionResult> CreateArt([FromBody] CreateArtRequestDto art)
     {
         return HandleResult(await Mediator.Send(new CreateArtCommand(art)));
     }
+
+    [HttpDelete]
+    public async Task<IActionResult> Delete([FromBody] DeleteArtRequestDto deleteArtRequestDto)
```
Comment (OlhaPivniuk) [inline Streetcode/Streetcode.WebApi/Controllers/Media/Images/ArtController.cs:39]: Why do not you use deleting by id? This way - [HttpDelete("{id:int}")]
public async Task<IActionResult> Delete([FromRoute] int id)

**Review by OlhaPivniuk — APPROVED** [review]

**Review by yaninavoda — COMMENTED** [review]
You shouldn't just remove art as it can be used by other streetcodes.
check if this art is not used in other streetcodes, then delete

**Review by yaninavoda — APPROVED** [review]

**Pull request merged**